gpcc-mocks/src/main/java/uk/nhs/adaptors/gpccmocks/gpc/GpcController.java

```diff
@@ -0,0 +1,208 @@
+package uk.nhs.adaptors.gpccmocks.gpc;
+
+import static org.springframework.http.HttpStatus.ACCEPTED;
+
+import static uk.nhs.adaptors.gpccmocks.common.ControllerHelpers.getHostAndPortFromRequest;
+import static uk.nhs.adaptors.gpccmocks.common.ControllerHelpers.getResponseHeaders;
+import static uk.nhs.adaptors.gpccmocks.common.ControllerHelpers.isUuid;
+import static uk.nhs.adaptors.gpccmocks.common.OperationOutcomes.badRequest;
+import static uk.nhs.adaptors.gpccmocks.common.OperationOutcomes.invalidNhsNumber;
+import static uk.nhs.adaptors.gpccmocks.common.OperationOutcomes.patientNotFound;
+import static uk.nhs.adaptors.gpccmocks.common.OperationOutcomes.referenceNotFound;
+
+import java.util.List;
+
+import javax.servlet.http.HttpServletRequest;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestHeader;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+import com.jayway.jsonpath.JsonPath;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import uk.nhs.adaptors.gpccmocks.common.TemplateUtils;
+
+@RestController
+@Slf4j
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
```

Comment:
You could add a part of the endpoint here which is similar to the other endpoints, like so
`@RequestMapping("/{odsCode}/STU3/1/gpconnect")`,
and then you can simply do
`@PostMapping(value = "/structured/fhir/Patient/$gpc.getstructuredrecord")`
`@GetMapping(value = "/documents/fhir/Patient/{patientId}/DocumentReference")`
etc

Reply:
Great suggestion!